test/PowerShellEditorServices.Test/Language/SemanticTokenTest.cs

```diff
@@ -0,0 +1,190 @@
+            );
+        }
+
+        [Fact]
+        public async Task RecognizesTokensWithAsterisk()
+        {
+            string text = @"
+function Get-A*A {
+}
+Get-A*A
+";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach (Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(t));
+                switch (t.Text)
+                {
+                    case "function":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Keyword, mappedTokens[0].Type));
+                        break;
+                    case "Get-A*A":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Function, mappedTokens[0].Type));
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        public async Task RecognizesArrayMemberInExpandableString()
+        {
+            string text = "\"$(@($Array).Count) OtherText\"";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach (Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(t));
+                switch (t.Text)
+                {
+                    case "$Array":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Variable, mappedTokens[0].Type));
+                        break;
+                    case "Count":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Member, mappedTokens[0].Type));
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        public async Task RecognizesCurlyQuotedString()
+        {
+            string text = "“^[-'a-z]*”";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(scriptFile.ScriptTokens[0]));
+            Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.String, mappedTokens[0].Type));
```

Comment:
If you're sure that mappedTokens has only one item in it, use this:
```suggestion
            Assert.Single(mappedTokens, sToken => Assert.Equal(SemanticTokenType.String, sToken.Type));
```